dodiscover/replearning/gin.py

```diff
@@ -0,0 +1,115 @@
+"""
+Wrapper for the GIN latent variable causal discovery algorithm in causal-learn
+"""
+
+from pandas import DataFrame
+
+from causallearn.search.HiddenCausal.GIN.GIN import GIN as GIN_
+from pywhy_graphs import CPDAG
+
+
+class GIN:
+    """Wrapper for GIN in the causal-learn package.
+
+    Parameters
+    ----------
+    indep_test_method : str
+        The method to use for testing independence, by default "kci"
+    alpha : float
+        The significance level for independence tests, by default 0.05
+
+    Attributes
+    ----------
+    graph_ : CPDAG
+        The estimated causal graph.
+    causal_learn_graph : CausalGraph
+        The causal graph object from causal-learn.
+    causal_ordering : list of str
+        The causal ordering of the variables.
+    """
+    def __init__(self, indep_test_method: str = "kci", alpha: float = 0.05):
+        """Initialize GIN object with specified parameters."""
+
+        self.graph_ = None
+
```

Comment:
You can probably remove this. The `_` character added at the end of a property name is usually "set" in the `fit/test` methods. This follows the design of scikit-learn, which we can then implement similar `check_is_fitted` utility functions.